spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/impl/AnalysisCache.java

```diff
@@ -238,7 +236,6 @@ public void dispose(){
     @Override
     @SuppressWarnings("unchecked")
     public <E> E getClassAnalysis(Class<E> analysisClass, @Nonnull ClassDescriptor classDescriptor) throws CheckedAnalysisException {
-        requireNonNull(classDescriptor, "classDescriptor is null");
```

Comment:
this will create NPE at line 249, where it will be not clear WHAT was null.

Reply:
@iloveeclipse AFAIK the NPE gets thrown immediately on method enter by the [Traute plugin](http://traute.oss.harmonysoft.tech/), assuming it's used and it works. Which may be a problem.

Reply:
Yes, an exception is thrown from an automatically generated check which is inserted into the method's beginning. The error message explicitly mentions target problem parameter's name.

Reply:
Stil I'm not convinced, may be too old school here.
The resulting stack trace will point to the code which does not exist, right? So given a bug report with some NPE we will need to guess which version of the code was used, because line numbers in the stack will show something no one can see in the code?

Also I'm worried about debugging in Eclipse - I guess the Traute compiler plugin isn't working if running Eclipse, because it looks like it is for javac only?

Means: running "same" code in Eclipse debugger (compiled by Eclipse ecj compiler) will show a different behavior? If yes, this is a real show stopper and I vote against this PR.

Reply:
> The resulting stack trace will point to the code which does not exist, right?

@iloveeclipse I guess, the stacktrace points to the method signature line, where there's no code, just the annotation. Someone might be confused one day, but the behavior is actually very logical and despite the possible confusion it's clear where to search for the culprit. So I don't think that this could be a problem (I'm not saying anything about the javac/ecj problem).